config/feature_flags.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package config
+
+import (
+	"math"
+	"reflect"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+	"github.com/splitio/go-client/v6/splitio/client"
+	"github.com/splitio/go-client/v6/splitio/conf"
+)
+
+type FeatureFlagSyncParams struct {
+	ServerID            string
+	SplitKey            string
+	SyncIntervalSeconds int
+	Log                 *mlog.Logger
+}
+
+type FeatureFlagSynchronizer struct {
+	FeatureFlagSyncParams
+
+	client  *client.SplitClient
+	stop    chan struct{}
+	stopped chan struct{}
+}
+
+func NewFeatureFlagSynchronizer(params FeatureFlagSyncParams) (*FeatureFlagSynchronizer, error) {
+	cfg := conf.Default()
+	if params.Log != nil {
+		cfg.Logger = &splitMlogAdapter{wrappedLog: params.Log.With(mlog.String("service", "split"))}
+	} else {
+		cfg.LoggerConfig.LogLevel = math.MinInt32
+	}
+	factory, err := client.NewSplitFactory(params.SplitKey, cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create split factory")
+	}
+
+	return &FeatureFlagSynchronizer{
+		FeatureFlagSyncParams: params,
+		client:                factory.Client(),
+		stop:                  make(chan struct{}),
+		stopped:               make(chan struct{}),
+	}, nil
+}
+
+// ensureReady blocks until the syncronizer is ready to update feature flag values
+func (f *FeatureFlagSynchronizer) EnsureReady() error {
+	if err := f.client.BlockUntilReady(10); err != nil {
+		return errors.Wrap(err, "split.io client could not initalize")
+	}
+
+	return nil
+}
+
+func (f *FeatureFlagSynchronizer) UpdateFeatureFlagValues(base model.FeatureFlags) model.FeatureFlags {
+	featureNames := getStructFields(model.FeatureFlags{})
```

Comment:
This is not going to change during the runtime of the program. We should compute this once in `NewFeatureFlagSynchronizer` and make it a struct member.

Reply:
Makes sense. Just made it a "const"